jjb/include/pipelines/qpc-tests-pipeline.groovy

```diff
@@ -0,0 +1,529 @@
+    archiveArtifacts "test-$testset-logs.tar.gz"
+    junit "$testset-junit.xml"
+}}
+
+
+def runCamayocUITest(browser) {{
+    echo "Running ${{testset}} Tests"
+
+    sh 'cat camayoc/config.yaml'
+    sh 'ls -lah'
+    sshagent(['390bdc1f-73c6-457e-81de-9e794478e0e']) {{
+        sh "sudo docker run --net='host' -d -p 4444:4444 -v /dev/shm:/dev/shm:z -v /tmp:/tmp:z selenium/standalone-$browser"
+
+        sleep 3
+
+        sh """\
+        export XDG_CONFIG_HOME=\$PWD
+        export SELENIUM_DRIVER=$browser
+
+        set +e
+        py.test -c pytest.ini -l -ra -vvv --junit-prefix $browser --junit-xml ui-$browser-junit.xml --rootdir camayoc/camayoc/tests/qpc camayoc/camayoc/tests/qpc/ui
+        set -e
+
+        sudo docker rm \$(sudo docker stop \$(sudo docker ps -aq))
+        tar -cvzf test-ui-$browser-logs.tar.gz log
+        sudo rm -rf log
+        """.stripIndent()
+    }}
+
+    echo 'Archiving artifacts'
+    archiveArtifacts "test-ui-$browser-logs.tar.gz"
+    junit "ui-$browser-junit.xml"
+}}
+
+
+
+////////////////////////
+//// Pipeline Stage ////
+////////////////////////
+stage('Run Tests') {{
+    parallel 'CentOS 7 Install': {{
+        node('centos7-os') {{
+            stage('Centos7 Install') {{
+                dir('ci') {{
+                    git 'https://github.com/quipucords/ci.git'
+                }}
+
+                sshagent(['390bdc1f-73c6-457e-81de-9e794478e0e']) {{
+                    withCredentials([file(credentialsId:
+                    '4c692211-c5e1-4354-8e1b-b9d0276c29d9', variable: 'ID_JENKINS_RSA')]) {{
+                        withEnv(['DISTRO=centos7', 'RELEASE={release}']) {{
+                            sh """\
+                            echo 'Testing qpc_version variable'
+                            echo ${{qpc_version}}
+                            """.stripIndent()
+                            echo "Testing inline qpc_version variable: ${{qpc_version}}"
+                            getQuipucords()
+                            installQPC 'centos7'
+                            runInstallTests 'centos7'
+                        }}
+                    }}
+                }}
+            }}
+            junit 'junit.xml'
+        }}
+    }},
+
+
+    'Fedora 28': {{
+        node('f28-os') {{
+            stage('Fedora 28 Install') {{
+                dir('ci') {{
+                    git 'https://github.com/quipucords/ci.git'
+                }}
+
+                sshagent(['390bdc1f-73c6-457e-81de-9e794478e0e']) {{
+                    withCredentials([file(credentialsId:
+                    '4c692211-c5e1-4354-8e1b-b9d0276c29d9', variable: 'ID_JENKINS_RSA')]) {{
+                        withEnv(['DISTRO=f28', 'RELEASE={release}']) {{
+                            echo 'Fedora 28: Configure Docker'
+                            setupDocker()
+                            getQuipucords()
+                            installQPC 'f28'
+//                            runInstallTests 'f28'
```

Comment:
Should this be dropped?

Reply:
I'm going to do some work on the install testing soon, and will plop this line back in once that's done. This comment is temporary.